bundles/binding/org.openhab.binding.fritzboxtr064/src/main/java/org/openhab/binding/fritzboxtr064/internal/FritzboxTr064Binding.java

```diff
@@ -0,0 +1,232 @@
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+	
+
+/**
+ * Implement this class if you are going create an actively polling service
+ * like querying a Website/Device.
+ * 
+ * @author Boris Bock 
+ * @since 0.1.0
+ */
+public class FritzboxTr064Binding extends AbstractActiveBinding<FritzboxTr064BindingProvider> {
+
+	private static final Logger logger = 
+		LoggerFactory.getLogger(FritzboxTr064Binding.class);
+
+	/**
+	 * The BundleContext. This is only valid when the bundle is ACTIVE. It is set in the activate()
+	 * method and must not be accessed anymore once the deactivate() method was called or before activate()
+	 * was called.
+	 */
+	private BundleContext bundleContext;
+	
+	//URL to connect to fbox. Provided in main cfg file
+	private String _url;
+	
+	//Username to use to connect to fbox
+	private String _user; 
+		
+	//PW
+	private String _pw; 
+	
+	/** 
+	 * the refresh interval which is used to poll values from the FritzboxTr064
+	 * server (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 60000;
+	
+	/***
+	 * holds Fbox connection
+	 */
+	private Tr064Comm _fboxComm = null;
+	
+	
+	public FritzboxTr064Binding() {
+	}
+		
+	
+	/**
+	 * Called by the SCR to activate the component with its configuration read from CAS
+	 * 
+	 * @param bundleContext BundleContext of the Bundle that defines this component
+	 * @param configuration Configuration properties for this component obtained from the ConfigAdmin service
+	 */
+	public void activate(final BundleContext bundleContext, final Map<String, Object> configuration) {
+		this.bundleContext = bundleContext;
+		logger.debug("FritzBox TR064 Binding activated!");
+			
+		// to override the default refresh interval one has to add a 
+		// parameter to openhab.cfg like <bindingName>:refresh=<intervalInMs>
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString)) {
+			refreshInterval = Long.parseLong(refreshIntervalString);
+			logger.debug("Custom refresh interval set to " + refreshInterval);
+		}
+
+		//Check if fritzbox parameters were provided in config, otherwise does not make sense going on...
+		String fboxurl = (String) configuration.get("url");
+		String fboxuser = (String) configuration.get("user");
+		String fboxpw = (String) configuration.get("pass");
+		if(fboxurl == null){
+			logger.error("Fritzbox URL was not provided in config. Shutting down binding.");
+			//how to shutdown??
+			setProperlyConfigured(false);
+			return;
+		}
+		if(fboxuser == null){
+			logger.error("Fritzbox User was not provided in config. Using default username.");
+		}
+		if(fboxpw == null){
+			logger.error("Fritzbox Password was not provided in config. Shutting down binding.");
```

Comment:
decrease to info these configuration errors just don't let the binding start which should not be treated as system error